ChangelogGenerator/ChangelogGenerator/ChangelogGenerator.cs

```diff
@@ -0,0 +1,195 @@
+            var creds = new Credentials(Options.GitHubToken);
+            GitHubClient.Credentials = creds;
+            ZenHubClient = new ZenHubClient(Options.ZenHubToken);
+        }
+
+        public async Task<string> GenerateChangelog()
+        {
+            string releaseId = await GetReleaseId();
+            Dictionary<IssueType, List<Issue>> issues = await GetIssuesByType(releaseId);
+            return GenerateMarkdown(releaseId, issues);
+        }
+
+        private async Task<string> GetReleaseId()
+        {
+            string[] repoParts = Options.Repo.Split('/');
+            Repository repo = await GitHubClient.Repository.Get(repoParts[0], repoParts[1]);
+            ZenHubRepositoryClient repoClient = ZenHubClient.GetRepositoryClient(repo.Id);
+
+            Response<ReleaseReport[]> releases = await repoClient.GetReleaseReportsAsync();
+
+            string releaseId = string.Empty;
+            foreach (var release in releases.Value)
+            {
+                if (release.Title == Options.Release)
+                {
+                    releaseId = release.ReleaseId;
+                    break;
+                }
+            }
+
+            if (releaseId == string.Empty)
+            {
+                throw new Exception($"No such release: {Options.Release}");
+            }
+            return releaseId;
+        }
+
+        private async Task<Dictionary<IssueType, List<Issue>>> GetIssuesByType(string releaseId)
+        {
+            Dictionary<IssueType, List<Issue>> issuesByType = new Dictionary<IssueType, List<Issue>>();
+
+            ZenHubReleaseClient releaseClient = ZenHubClient.GetReleaseClient(releaseId);
+            IssueDetails[] zenHubIssueList = (await releaseClient.GetIssuesAsync()).Value;
+            string[] repoParts = Options.Repo.Split('/');
+            foreach (IssueDetails details in zenHubIssueList)
+            {
+                Issue issue = await GitHubClient.Issue.Get(repoParts[0], repoParts[1], details.IssueNumber);
+                bool issueFixed = true;
+                bool hidden = false;
+                IssueType issueType = IssueType.None;
+                bool epicLabel = false;
+                bool regressionDuringThisVersion = false;
+                bool engineeringImprovement = false;
+                string requiredLabel = Options.RequiredLabel?.ToLower();
+                bool foundRequiredLabel = string.IsNullOrEmpty(requiredLabel);
+
+                foreach (var label in issue.Labels)
+                {
+                    if (label.Name.Contains("ClosedAs:"))
+                    {
+                        issueFixed = false;
+                    }
+
+                    if (label.Name == "RegressionDuringThisVersion")
+                    {
+                        regressionDuringThisVersion = true;
+                        hidden = true;
+                    }
+
+                    if (label.Name == "Category:Engineering")
```

Comment:
Can we extract these labels to a file with statics?
Specifically my concern is what happens when my label cleanup ideas get adopted :D 
Would make it easier to fix up the tool.

Reply:
This is a good plan. I'll do that.